lib/src/scene.dart

```diff
@@ -71,4 +71,21 @@ abstract class Scene {
   ///
   /// This is called on every rebuild, including by Hot Reload.
   Widget build();
+
+  /// Invoke this to force Stager to rebuild the current [Scene]. Stager assigns
+  /// a value to this property – you should never need to set it. You will most
+  /// likely only need this if you override [environmentControlBuilders].
+  VoidCallback rebuildScene = () {};
```

Comment:
Is this style of dependency injection (having the owner essentially just set a dependency on a mutable property) common? The fact that it's unfamiliar to me irks me a bit, but I think the ergonomics are overall excellent because people are so familiar with the setState pattern. I can't really think of a better option, other than to have environmentControlBuilders actually be a function that accepts this as a parameter. E.g.

List<WidgetBuilder> createEnvironmentControlBuilders(VoidCallback rebuildScene) {
  return <WidgetBuilder>[];
}

That only seems harder to understand though. What do you think?

Reply:
Yeah, I'm not in love with the `rebuildScene` approach I have here for a couple reasons:

1. I think it would be _incredibly_ easy to forget to call `rebuildScene()`
2. When the user forgets to call `rebuildScene()`, the reason that their custom controls aren't working won't be obvious.

Idea: What if, instead of `environmentControlBuilders` being a list of `WidgetBuilder`s, I use something like `typedef EnvironmentControlBuilder = void Function(WidgetBuilder builder, VoidCallback rebuildScene)`? It makes the `environmentControlBuilders` a bit more verbose, but this way there's not a "hidden" function that's easy to forget and difficult to discover.

Reply:
Implemented this in 6ac23ce15892a04c55b2dcf6a2ea911494f7a193 and like it a lot better. lmk what you think!

Reply:
I like it! Thanks!